### PR TITLE
Issue 1216 part-1: Add client access to segment attributes

### PR DESCRIPTION
**Pull request by tkaitchuck**
**Change log description**
* Adds GetSegmentAttribute and UpdateSegmentAttribute wire commands and server support. (Note this is a backwards compatible wire format change.)
* Exposes SegmentMetadata.NULL_ATTRIBUTE_VALUE to the client by duplicating the value in WireCommand.

**Purpose of the change**
This adds wire-protocol support for clients to use Attributes. This is a prerequisite for #1216 

**What the code does**
Adds new requests and replies that are handled by PravegaRequestProcessor

**How to verify it**
Tests are added for serialization, and for the new processor methods.

### CODE REVIEW

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java [hunk 1]
```diff
@@ -205,8 +215,55 @@ private ByteBuffer copyData(List<ReadResultEntryContents> contents) {
...
+        String segmentName = updateSegmentAttribute.getSegmentName();
+        UUID attributeId = updateSegmentAttribute.getAttributeId();
+        long newValue = updateSegmentAttribute.getNewValue();
+        long expectedValue = updateSegmentAttribute.getExpectedValue();
+        val update = new AttributeUpdate(attributeId, AttributeUpdateType.ReplaceIfEquals, newValue, expectedValue);
+        segmentStore.updateAttributes(segmentName, Collections.singletonList(update), TIMEOUT).whenComplete((v, e) -> {
+            if (e == null) {
+                connection.send(new SegmentAttributeUpdated(requestId, true));
+            } else {
+                if (ExceptionHelpers.getRealException(e) instanceof BadAttributeUpdateException) {
```
Comment (andreipaduroiu) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java]: I just want to check that this exception is actually logged somewhere. I know that for a while we were not really logging server-side exceptions caused by bad input.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java [hunk 2]
```diff
@@ -205,8 +215,55 @@ private ByteBuffer copyData(List<ReadResultEntryContents> contents) {
...
+                    connection.send(new SegmentAttributeUpdated(requestId, false));
+                } else {
+                    handleException(requestId, segmentName, "Update attribute", e);
+                }
+            }
+        }).exceptionally((Throwable e) -> {
+            handleException(requestId, segmentName, "Update attribute", e);
```
Comment (andreipaduroiu) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java]: Can this be included in whenComplete? Or is it the catch block for the whenComplete arg?

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java (hunk 1 above)
Comment (tkaitchuck) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java]: I added a debug log.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java (hunk 2 above)
Comment (tkaitchuck) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java]: This is the catch block for the when complete. 4 lines above you can see it is invoking this same method if there is an exception returned from the call.

**Pull request merged**